File: document-ui/src/main/java/dev/sayaya/handbook/client/domain/Validation.java
package dev.sayaya.handbook.client.domain;

import lombok.Builder;
import lombok.Data;
import lombok.Singular;
import lombok.experimental.Accessors;

import java.util.Map;

@Data
@Accessors(fluent = true)
@Builder(toBuilder = true)
public class Validation {
    @Singular("value")
    private Map<String, Boolean> values;
    @Builder.Default
    private ValidationState state = ValidationState.DONE;

    public enum ValidationState {
        DONE
    }
}

File: domain/src/main/kotlin/dev/sayaya/handbook/domain/Validation.kt
package dev.sayaya.handbook.domain

import java.io.Serializable

@JvmRecord
data class Validation (
    val status: Status,
    val result: Map<String, Boolean>?
): Serializable {
    companion object {
        enum class Status {
            NEW,        // 작업이 생성되어 처리 대기 중
            PROCESSING, // 작업이 워커에 의해 처리 중
            DONE,       // 작업 성공적으로 완료
            FAILED      // 작업 실패
        }
    }
}